core/tls-websocket-server/src/tls_common.rs

```diff
@@ -41,42 +35,26 @@ pub fn make_config(cert: &str, key: &str) -> WebSocketResult<Arc<rustls::ServerC
 	Ok(Arc::new(config))
 }
 
-fn load_certs(filename: &str) -> WebSocketResult<Vec<rustls::Certificate>> {
-	let certfile = fs::File::open(filename).map_err(|e| {
-		WebSocketError::InvalidCertificate(format!(
-			"Failed to load certificate from file ({}): {:?}",
-			filename, e
-		))
-	})?;
-	let mut reader = BufReader::new(certfile);
-	rustls::internal::pemfile::certs(&mut reader).map_err(|_| {
-		WebSocketError::InvalidCertificate("Failed to parse certificate file".to_string())
-	})
+fn load_certs(content: &str) -> WebSocketResult<Vec<rustls::Certificate>> {
+	let mut reader = BufReader::new(content.as_bytes());
+	rustls::internal::pemfile::certs(&mut reader)
+		.map_err(|_| WebSocketError::InvalidCertificate("Failed to parse certificate".to_string()))
 }
 
-fn load_private_key(filename: &str) -> WebSocketResult<rustls::PrivateKey> {
+fn load_private_key(content: &str) -> WebSocketResult<rustls::PrivateKey> {
 	let rsa_keys = {
-		let keyfile = fs::File::open(filename).map_err(|e| {
-			WebSocketError::InvalidPrivateKey(format!(
-				"Failed to load private key from file ({}): {:?}",
-				filename, e
-			))
-		})?;
-		let mut reader = BufReader::new(keyfile);
+		let mut reader = BufReader::new(content.as_bytes());
 
 		rustls::internal::pemfile::rsa_private_keys(&mut reader).map_err(|_| {
```

Comment:
How can we differentiate rsa key and pkcs8 key if we don't use the pem format here?

Reply:
If we could have switched to `DER` only format, then I think the [`rustls::sign::any_supported_type`](https://docs.rs/rustls/latest/rustls/sign/fn.any_supported_type.html) would have worked as well.

Reply:
OK. Thanks